### DISCUSSION TITLE
Strange error building React Native iOS project

### DISCUSSION THREAD

**Discussion by brutaldev** [codemagic-ci-cd/codemagic-docs#1202] [Q&A] [answered]
The following error suddenly appeared for a React Native iOS build:

`xcode-project: error: argument --export-options-plist: Path "/Users/builder/export_options.plist" does not exist`

![image](https://user-images.githubusercontent.com/1929531/204906068-e2de12e3-9cf6-47d0-be0e-a8ce2369cab4.png)

That file does not exist in our project at all so not sure why it's expected to be there?

**Comment by himesh-cm** [comment, marked as answer]
Hi @brutaldev - are you running the below script before the build command?
```
- name: Set up provisioning profiles settings on Xcode project
        script: xcode-project use-profiles
```

**Reply by brutaldev** [reply]
Hi @himesh-cm 
Was definitely missing that part and found that from a mention that `export_options` comes from the `user-profiles` command. However, that did not quite solve it because it appears as though that command is correctly finding the profile to use which is an ad-hoc profile and certificate but somehow thinks it's a distribution signing certificate as well and things just get confusing from there.

Output from the `Set up provisioning profiles settings on Xcode project` step:
```
Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/30-11-22_fc4ce33m.keychain-db
Searching for files matching /Users/builder/clone/ios/AwesomeProject.xcodeproj
Completed configuring code signing settings
 - Using profile "XYZ Demo" [54e59d5b-a884-4cb0-b4c8-3fd3bb7ef285] for target "AwesomeProject" [Debug] from project "AwesomeProject"
 - Using profile "XYZ Demo" [54e59d5b-a884-4cb0-b4c8-3fd3bb7ef285] for target "AwesomeProject" [Release] from project "AwesomeProject"
Generated options for exporting the project
 - Method: ad-hoc
 - Provisioning Profiles:
     - com.notreal.demo: XYZ Demo
 - Signing Certificate: Apple Distribution    <---- What??
 - Signing Style: manual
 - Team Id: 123456ABCDEF
Saved export options to /Users/builder/export_options.plist
```

After that the build breaks with more weird errors:
```
    Run script build phase 'Start Packager' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'AwesomeProject' from project 'AwesomeProject')
    Run script build phase 'Bundle React Native code and images' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'AwesomeProject' from project 'AwesomeProject')
    Run script build phase '[CP-User] [RNFB] Core Configuration' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'AwesomeProject' from project 'AwesomeProject')
Failed to archive /Users/builder/clone/ios/AwesomeProject.xcworkspace

Build failed :|
Step 7 script `Build ipa for distribution` exited with status code 65
```
If you have a quick answer for that it would be great, otherwise you were correct that the `Set up provisioning profiles settings on Xcode project` step was missing causing the error that I reported.

**Reply by himesh-cm** [reply]
Hey @brutaldev,  sure, please share your build id to investigate further.

**Reply by brutaldev** [reply]
Here we go: 6387c6358558d4e42e4d3af8
I've recreated all the certs and made new provisioning profiles just to make sure and it's still happening.

**Reply by brutaldev** [reply]
@himesh-cm Only breaks when the method is `ad-hoc`.
If using a different profile (same distribution cert) then it works perfectly.
Here is the same build setup but using the store distribution profile: 6387cf0c59ce355448768434

**Reply by himesh-cm** [reply]
Hi @brutaldev - when you are building with Adhoc Profile - Add the following to the end of your xcode-project build-ipa command:
```
  --archive-flags="-destination 'generic/platform=iOS'"
```

**Reply by brutaldev** [reply]
Got it, thanks for the help!
